src/controller/CurrentFabricRemover.h

```diff
@@ -0,0 +1,107 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <app/OperationalSessionSetup.h>
+#include <controller/CHIPDeviceController.h>
+#include <lib/core/CHIPCallback.h>
+
+namespace chip {
+namespace Controller {
+
+typedef void (*OnCurrentFabricRemove)(void * context, NodeId remoteDeviceId, CHIP_ERROR status);
+
+/**
+ * A helper class to remove fabric given some parameters.
+ */
+class CurrentFabricRemover
+{
+public:
+    CurrentFabricRemover(DeviceCommissioner * commissioner) :
```

Comment:
Is there any reason we can't have fabric removal with a DeviceController?

Reply:
@msandstedt See https://github.com/project-chip/connectedhomeip/pull/24127#issuecomment-1361460170 and preceding discussion.

Reply:
I will change to DeviceController parameter.